contrib/vagrant/provision-minimal.sh

```diff
@@ -1,20 +1,64 @@
 #!/bin/bash
-set -euo pipefail
-IFS=$'\n\t'
 
-sed -i s/^Defaults.*requiretty/\#Defaults\ requiretty/g /etc/sudoers
+function setup {
+	set -euo pipefail
+	IFS=$'\n\t'
 
-# patch incompatible with fail-over DNS setup
-SCRIPT='/etc/NetworkManager/dispatcher.d/fix-slow-dns'
-if [[ -f "${SCRIPT}" ]]; then
-    echo "Removing ${SCRIPT}..."
-    rm "${SCRIPT}"
-    sed -i -e '/^options.*$/d' /etc/resolv.conf
-fi
-unset SCRIPT
+	sed -i s/^Defaults.*requiretty/\#Defaults\ requiretty/g /etc/sudoers
 
-if [ -f /usr/bin/generate_openshift_service ]
-then
-  sudo /usr/bin/generate_openshift_service
-fi
+	# patch incompatible with fail-over DNS setup
+	SCRIPT='/etc/NetworkManager/dispatcher.d/fix-slow-dns'
+	if [[ -f "${SCRIPT}" ]]; then
+	    echo "Removing ${SCRIPT}..."
+	    rm "${SCRIPT}"
+	    sed -i -e '/^options.*$/d' /etc/resolv.conf
+	fi
+	unset SCRIPT
 
+	if [ -f /usr/bin/generate_openshift_service ]
+	then
+	  sudo /usr/bin/generate_openshift_service
+	fi
+}
+
+function installOpenShift {
```

Comment:
ah ok.  i thought the CDK was used just by app developers.  i didnt realize it was a tool for developing OS itself... ?  i do recall you mentioned it in the other issues as well.